src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -221,6 +209,12 @@ protected function buildDeployment(Device $deviceForFunctions, Func $queueForFun
                 $stdout = '';
                 $stderr = '';
                 Console::execute('mkdir -p /tmp/builds/' . \escapeshellcmd($buildId), '', $stdout, $stderr);
+
+                $build = $dbForProject->getDocument('builds', $buildId);
+                if ($build->getAttribute('status') === 'cancelled') {
+                    return;
```

Comment:
Is the return sufficient? Can we also log that the build has been cancelled and then return?

Reply:
Added console::info